php/libraries/Module.class.inc

```diff
@@ -150,7 +153,7 @@ class Module extends \LORIS\Router\PrefixRouter implements RequestHandlerInterfa
      * Returns true if this module should be publicly available to people who
      * are not logged in to LORIS.
      *
-     * @return bool (false: not that this is not currently possible because main.php
+     * @return bool (false: note that this is not currently possible because main.php
      *         checks that the user is logged in before it calls NDB_Caller.)
      */
     public function isPublicModule()
```

Comment:
I wasn't able to add a return type here without offending `phan`. Thsi signature must match the function in `modules/api/php/module.class.inc` but when both are delcared to return bool I get the following error.
```
modules/login/php/module.class.inc:36 PhanParamSignatureRealMismatchReturnType Declaration of function isPublicModule() should be compatible with function isPublicModule() : bool (method returning '' cannot override method returning 'bool') defined in php/libraries/Module.class.inc:159
```

I wasn't able to fix it.

Reply:
That sounds like the login module class (and likely all the module's module classes) needs to be updated to have the same signature as the parent class.